pkg/provision/buildroot.go

```diff
@@ -72,6 +72,8 @@ Type=notify
 Environment=DOCKER_RAMDISK=yes
 {{range .EngineOptions.Env}}Environment={{.}}
 {{end}}
+
+ExecStart=
```

Comment:
Right -- the first `ExecStart=` clears the entry inherited from the base dockerd configuration. Without this, the `ExecStart` on line 77 would be treated as the *second* command to execute for the service (after executing the command specified in the base config), which systemd rejects as invalid for this type of service.

As you noticed, this change was motivated by a change ([docker/machine#3966](https://github.com/docker/machine/pull/3966)) in docker/machine to inherit from instead of overwriting the base dockerd config.